client/src/client.rs

```diff
@@ -480,18 +493,9 @@ impl Client {
 
     // TODO: introduce timeout
     fn listen_for_tx_confirmation(
-        &self,
+        event_iterator: impl Iterator<Item = Result<Event, eyre::ErrReport>>,
         hash: HashOf<SignedTransaction>,
-        init_sender: &mpsc::Sender<()>,
     ) -> Result<HashOf<VersionedSignedTransaction>> {
-        let event_iterator = self
-            .listen_for_events(PipelineEventFilter::new().hash(hash.into()).into())
-            .wrap_err("Failed to establish event listener connection")?;
-
-        init_sender
-            .send(())
-            .wrap_err("Failed to send init message through init channel")?;
-
         for event in event_iterator.flatten() {
```

Comment:
This place could be considered harmful too, because we are ignoring all errors and possibly stucked in infinite loop.